### PR TITLE
updates rate limit docs to remove mention of token specific rate limit

### PR DESCRIPTION
**Pull request by myblake**
After https://github.com/Finch-API/api-server/pull/6747 was merged we no longer enforce token level rate limits. This change updates the docs to solely focus on application limits.

### CODE REVIEW

File: docs/Best-Practices/Handling-Rate-Limit-Errors.md [hunk 1]
```diff
@@ -164,41 +102,23 @@ In this scenario, let's assume your application has six access tokens (Token A,
...
+1. Token C, D, and E repeats the same process as Token B making 5 requests to the [/employer/company](https://developer.tryfinch.com/docs/reference/33162be1eed72-company) endpoint, 3 requests to the [/employer/directory](https://developer.tryfinch.com/docs/reference/12419c085fc0e-directory) endpoint, and 2 requests to the [/employer/payment](https://developer.tryfinch.com/docs/reference/b811fdc2542ca-payment) endpoint all within the same minute. The requests will succeed until the 20th request to the  [/employer/company](https://developer.tryfinch.com/docs/reference/33162be1eed72-company) endpoint, after which point each token fails and returns a 429 rate limit error.
     - Application-level rate limits
         Bucket | Capacity
         -------|-------------
         `company` | 20/20 (FULL)
         `directory` | 15/20
         `individual` | 0/20
         `employment` | 0/20
-        `payment` | 12/12 (FULL)
+        `payment` | 10/12
```
Comment (myblake) [inline docs/Best-Practices/Handling-Rate-Limit-Errors.md:113]: Turns out the docs also did some suspiciously bad addition for the payment endpoint in this scenario 🙄 

https://developer.tryfinch.com/docs/reference/b288acfbb6816-handling-rate-limit-errors

Comment (aren55555) [reply docs/Best-Practices/Handling-Rate-Limit-Errors.md:113]: Nice catch -- love the attn to detail

File: docs/Development-Guides/Rate-Limits.md [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 Finch will return a rate limit error with the HTTP status code 429 when the request rate limit for an application, access token, or an individual IP address has been exceeded.
 
-Finch's rate limits work on a per-product basis for both applications and access tokens. Rate limits are summed on a rolling 60-second basis for each unique `product`. This is commonly referred to as a Sliding or Rolling Window rate limit.
+Finch's rate limits work on a per-product basis for applications. Rate limits are summed on a rolling 60-second basis for each unique `product`. This is commonly referred to as a Sliding or Rolling Window rate limit.
```
Comment (aren55555) [inline docs/Development-Guides/Rate-Limits.md:5]: I'm not sure of the desired nomenclature but "per-product" might be contextually easier to understand if we just said "per-endpoint" or "per-path".

**Review by aren55555 — APPROVED** [review]

File: docs/Development-Guides/Rate-Limits.md (hunk 2 above)
Comment (myblake) [reply docs/Development-Guides/Rate-Limits.md]: Good catch, this was confusing and I've updated it.

**Pull request merged**